### PR TITLE
qucs-s: 0.0.22 -> 1.0.2

### PR DESCRIPTION
**Pull request by evilmav**
###### Description of changes

For changelog see [Releases](https://github.com/ra3xdh/qucs_s/releases) for 0.0.23 .. 1.0.1.
Qt has been bumped from 4 to 6 as 5 is approaching EOL.

Checked the program opens, a simple circuit can be drown and ngspice integration works.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ajs124** [comment]
For reference, I did something similar in https://github.com/NixOS/nixpkgs/pull/174634/commits/c2abb8638a3a22619fd63c29e0418f8697c09d2a. This PR probably makes that commit obsolete.

**Review by Tungsten842 — CHANGES_REQUESTED** [review]

**Review by Tungsten842 — APPROVED** [review]
It works fine, tested with some simple circuits.

File: pkgs/applications/science/electronics/qucs-s/default.nix [hunk 1]
```diff
@@ -1,27 +1,43 @@
...
     repo = "qucs_s";
     rev = version;
-    sha256 = "0rrq2ddridc09m6fixdmbngn42xmv8cmdf6r8zzn2s98fqib5qd6";
+    sha256 = "sha256-IdIq/ZY8y/OJ2E51DwhJvcgT+cMAMLozgSuxn8As6ZY=";
   };
 
-  nativeBuildInputs = [ wrapGAppsHook cmake ];
-  buildInputs = [ flex bison qt4 libX11 gperf adms ] ++ kernels;
+  nativeBuildInputs = [ wrapQtAppsHook cmake ];
+  buildInputs = [ flex bison qtbase qttools qtsvg qtwayland libX11 gperf adms ] ++ kernels;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/qucs-s/default.nix:31]: flex and bison go to nativeBuildInputs most likely

**Comment by Luz** [comment]
Works for me.
(I used ngspice 37 for testing, as ngspice 38 is still in progress, https://github.com/NixOS/nixpkgs/pull/199545 )

File: pkgs/applications/science/electronics/qucs-s/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/science/electronics/qucs-s/default.nix:31]: this muts be moved to the correct input

Comment (EvilMav) [reply pkgs/applications/science/electronics/qucs-s/default.nix]: Thank you for the hint! I've changed it [the next commit](https://github.com/NixOS/nixpkgs/pull/223378/commits/1ad5d99ecf15e6825579a904f7c92a0ddc39d174) while bumping to 1.0.2.

**Comment by EvilMav** [comment]
Bumped to 1.0.2. @Luz can you please check it still finds the ngspice by default for you? For some reason mine does not find `ngspice` command, which should be on the PATH as ensured by the wrapper... Judging by strace, it does not try the PATH:
```
$ strace ./result/bin/qucs-s 2> strace.txt  
# run a simulation
$ cat strace.txt | grep ngspice                                                                                                                                                                                                                                                                                                             
openat(AT_FDCWD, "/nix/store/6l00n6isj6b70yfg02vki4676i7xg15f-qucs-s-1.0.2/share/qucs-s/library//ngspice.blacklist", O_RDONLY|O_CLOEXEC) = 12
openat(AT_FDCWD, "/nix/store/6l00n6isj6b70yfg02vki4676i7xg15f-qucs-s-1.0.2/share/qucs-s/library//ngspice.blacklist", O_RDONLY|O_CLOEXEC) = 12
access("/nix/store/6l00n6isj6b70yfg02vki4676i7xg15f-qucs-s-1.0.2/bin/ngspice", F_OK) = -1 ENOENT (No such file or directory)
access("/nix/store/6l00n6isj6b70yfg02vki4676i7xg15f-qucs-s-1.0.2/bin/ngspice", F_OK) = -1 ENOENT (No such file or directory)
access("/nix/store/6l00n6isj6b70yfg02vki4676i7xg15f-qucs-s-1.0.2/share/qucs-s/xspice_cmlib/include/ngspice_mathfunc.inc", F_OK) = 0
openat(AT_FDCWD, "/home/maverick/Desktop/test-qucs.dat.ngspice", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 19
access("/nix/store/6l00n6isj6b70yfg02vki4676i7xg15f-qucs-s-1.0.2/share/qucs-s/xspice_cmlib/include/ngspice_mathfunc.inc", F_OK) = 0
openat(AT_FDCWD, "/home/maverick/Desktop/test-qucs.dat.ngspice", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666) = 19
```

**Comment by Luz** [comment]
For me it works!

What I did:
1. Install ngspice (edit my configuration.nix, then nixos-rebuild switch)
2. Verify that it is available by typing "ngspice" in the console
3. nix-build -A qucs-s
4. Execute binary: /nix/store/7zj3dla8nqggfgk98viiy0wvlc0sx8pm-qucs-s-1.0.2/bin/qucs-s
5. Copied the example "/nix/store/7zj3dla8nqggfgk98viiy0wvlc0sx8pm-qucs-s-1.0.2/share/qucs-s/examples/ngspice/audio_amp.sch" to my workspace. (It was important to use the official example, as my selfmade simple example failed to run... The tool said something like "ngspice can not be executed", which is probably a misleading error message from qucs-s.)
6. Opened it and ran by "F2"
Also When I look at "Simulation -> Select default simulator" I see that "ngspice" is selected as "default simulator" and "ngspice" is the text in the field about the "ngspice executable location".

Thanks for your work!

**Review by Luz — APPROVED** [review]

**Pull request merged**